src/modules/roc_audio/profiling_reader.cpp

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include "roc_audio/profiling_reader.h"
+#include "roc_core/log.h"
+#include "roc_core/panic.h"
+
+namespace roc {
+namespace audio {
+
+ProfilingReader::ProfilingReader(IReader& reader,
+                                 core::IAllocator& allocator,
+                                 packet::channel_mask_t channels,
+                                 size_t sample_rate)
+    : profiler_(allocator, channels, sample_rate)
+    , reader_(reader) {
+}
+
+void ProfilingReader::read(Frame& frame) {
+    profiler_.begin_frame(frame);
+
+    const core::nanoseconds_t elapsed = read_(frame);
+
+    profiler_.end_frame(frame, elapsed);
+}
+
+core::nanoseconds_t ProfilingReader::read_(Frame& frame) {
+    const core::nanoseconds_t start = core::timestamp();
```

Comment:
Though, I you'll add unit tests for Profiler (https://github.com/roc-project/roc/pull/291#issuecomment-562073898), it probably makes sense to keep this as is, so that Profiler will not depend on real time and can be tested easily.